### PR TITLE
Add SHOW FUNCTIONS support to client

### LINKED ISSUE #52757: SHOW FUNCTIONS and SHOW FUNCTIONS LIKE
**Use case**

Quickly type it in clickhouse-client.

**Additional context**

It could be identical to:
`SELECT * FROM system.functions` and `SELECT * FROM system.functions WHERE name LIKE '...'`
but please also take a look at how it works in MySQL, Postgres, and SQLite.

### PR DESCRIPTION
**Pull request by jkartseva**
Accepts CH client queries in the form of
SHOW FUNCTIONS [LIKE | ILIKE '< pattern>']

Retrieves all columns from `functions` table in system database.

Fixes #52757

### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add SHOW FUNCTIONS support to clickhouse-client

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
This is an automated comment for commit a7bda944b3bb035ea7156d3f5e34d2fee0efbef9 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/54337/a7bda944b3bb035ea7156d3f5e34d2fee0efbef9/ci_running.html)
The overall status of the commit is 🔴 failure

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🔴 failure</td></tr>
</table>

**Comment by Avogar** [comment]
Please, add documentation about this new `SHOW` statement in [docs/en/sql-reference/statements/show.md](https://github.com/ClickHouse/ClickHouse/blob/master/docs/en/sql-reference/statements/show.md)

File: src/Parsers/ParserShowFunctionsQuery.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#include <Parsers/ASTShowFunctionsQuery.h>
+#include <Parsers/CommonParsers.h>
+#include <Parsers/ExpressionElementParsers.h>
+
+namespace DB
+{
+
+bool ParserShowFunctionsQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ASTPtr database;
```
Comment (Avogar) [inline src/Parsers/ParserShowFunctionsQuery.cpp:13]: This variable is unused, let's delete it

File: tests/queries/0_stateless/00419_show_sql_queries.sh [hunk 2]
```diff
@@ -8,3 +8,4 @@ $CLICKHOUSE_CLIENT -q "SHOW PROCESSLIST" &>/dev/null
 $CLICKHOUSE_CLIENT -q "SHOW DATABASES" &>/dev/null
 $CLICKHOUSE_CLIENT -q "SHOW TABLES" &>/dev/null
 $CLICKHOUSE_CLIENT -q "SHOW ENGINES" &>/dev/null
+$CLICKHOUSE_CLIENT -q "SHOW FUNCTIONS" &>/dev/null
```
Comment (Avogar) [inline tests/queries/0_stateless/00419_show_sql_queries.sh:11]: I think this test is not enough, it doesn't test the correctness of the result of the query and it doesn't test `LIKE/ILIKE` clauses. Let's also create a separate test for this

**Review by Avogar — APPROVED** [review]
Looks great! Thank you for the contribution!

**Pull request merged**